### PR TITLE
Added draft for DEP 0101 clarifying release process.

### PR DESCRIPTION
**Pull request by carltongibson**

### CODE REVIEW

**Comment by carltongibson** [comment]
Hey @felixxm — can I ask for your initial 👀 before I ask for wider review and input? Thanks! 🎁

File: draft/0101-release-process.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+Scheduling is entirely mechanical. Django has a fixed eight monthly release
+cycle for major releases — Apr - Dec - Aug, and back to Apr over a 24month
+period.
+
+Final releases are made early in the first week of the month. Major releases
+have a pre-release phase, which has one alpha release, one beta, and one
+release candidate.
+
+* The release candidate comes two weeks before the final release.
+* The beta comes one month before the release candidate.
```
Comment (felixxm) [inline draft/0101-release-process.rst:51]:
```suggestion
* The beta comes four weeks before the release candidate.
```

File: draft/0101-release-process.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+* The alpha comes one month before the beta.
```
Comment (felixxm) [inline draft/0101-release-process.rst:52]:
```suggestion
* The alpha comes five weeks before the beta.
```

File: draft/0101-release-process.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+The exact day is left to the Releaser's discretion.
+
+.. note::
+
+    Factors influencing exact release day
+
+    Factors that are considered when setting the exact release day include:
+    being as close to the 1st of the month as possible, releaser availability,
+    avoiding Fridays and weekends, avoiding days such as April 1st, and the
+    largest international public holidays, such as New Years day.
```
Comment (felixxm) [inline draft/0101-release-process.rst:63]:
```suggestion
    largest international public holidays, such as New Year's Day.
```

File: draft/0101-release-process.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+Within this bugfix and security releases occur at the beginning on each month
+for each stable version with applied changes to be released during its
+supported lifetime.
+
+Finally, in exceptional circumstances, additional releases can be made outside
+of the release schedule. Historical examples include a security release made
+quickly due to a high severity vulnerability being made public, and an ad hoc
+release made to correct a packaging error.
+
+As per DEP 4, release may be held back if there are unresolved release blockers.
```
Comment (felixxm) [inline draft/0101-release-process.rst:89]: Maybe we can add that scheduled security releases are not blocked by unresolved release blockers.

File: draft/0101-release-process.rst [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+desire to change the release process exists, a separate DEP making that case
+can be presented.
+
+On specific points of the original DEP 10 phrasing on the release process:
+
+* The release schedule is set entirely mechanically, as per the specification
+  above. There is no determination needed, and so the requirement for the
+  technical board to do so introduces artificial busy-work.
+* Whilst the technical board can raise a flag if necessary, in the normal case,
+  releases procede automatically, on-schedule, unless there is a specific
```
Comment (felixxm) [inline draft/0101-release-process.rst:141]:
```suggestion
  releases proceed automatically, on-schedule, unless there is a specific
```

File: draft/0101-release-process.rst [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  reason, such as an open release blocker, for them not to. The introduction of
+  the requirement for an approval vote before release introduces a risk of a
+  release not occurring for a procedural failing if a vote, under DEP10's quite
+  strict voting procedures, is not successfully held.
+* Releasing a major version of Django is an extremely stressful activity. It
+  already has many moving parts. It takes the full effort of the Releaser
```
Comment (felixxm) [inline draft/0101-release-process.rst:147]:
```suggestion
  already has many moving parts. It takes the full effort of the release manager
```

File: draft/0101-release-process.rst [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+  on-hand, likely with assistance from other Releasers, to do everything
+  correctly. Adding the requirement to ensure that a technical board vote is
+  held, when this is in-truth merely rubber-stamping the release, is an
+  unnecessary extra burden.
+
+In a utopian world where technical board members had the time and capacity to
```
Comment (felixxm) [inline draft/0101-release-process.rst:153]: Maybe `utopian` -> `perfect` :wink:

**Review by felixxm — COMMENTED** [review]
@carltongibson Thanks :+1:

File: draft/0101-release-process.rst [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  correctly. Adding the requirement to ensure that a technical board vote is
+  held, when this is in-truth merely rubber-stamping the release, is an
+  unnecessary extra burden.
+
+In a perfect world where technical board members had the time and capacity to
+be more directly involved in the day-to-day development of Django, these last
+two points concerning the vote-to-release flow would perhaps be minimised.
+That's not our world, however. Even if it were though, the correct procedure is
+to not put potential pitfalls on the default path. We assume the release goes
+ahead, unless there's a reason not to.
```
Comment (carltongibson) [inline draft/0101-release-process.rst:159]: I think these points can be dropped as I don't think the DEP 10 wording in fact requires vote-to-release in normal circumstances. 

See https://github.com/django/deps/pull/77/files#r1012854301

File: draft/0101-release-process.rst [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+
+Since its introduction in 2015, the time-based release process, and the manner
+in which it ties into the LTS, API stability, and deprecation policies has been
+an unmitigated success. It has brought key benefits of predictability and ease
+of updates.
+
+Any change to Django's release process would require a full DEP of its own and
+it was not part of DEP 10's scope to address this.
+
+DEP 10 concerned the dissolving of the old Django Core, the establishment or
```
Comment (carltongibson) [inline draft/0101-release-process.rst:125]:
```suggestion
DEP 10 concerned the dissolving of the old Django Core, the establishment of
```

**Review by apollo13 — APPROVED** [review]
The DEP looks good to me. I don't have any strong feelings on the paragraph/bullet points you want to strike out. I agree with you (at least in the way I read DEP 10) that there is no strong requirement for the technical board to hold a vote about the release, especially given our established release process I see the technical board more of in an oversight role here. While this may not be in line with what was written down (but we are now correcting that), I think it was in the spirit of DEP 10 to not destroy the existing (working) process.

**Comment by apollo13** [comment]
While I do approve the suggested changes here, I am wondering if it might make sense to trim DEP 10 down on the release process, put the specifics solely into DEP 4 (or a new DEP 44 or similar) and refer to that? Do we have any strong rules on changing deps? Changing DEP 4 which is final obviously doesn't seem great, on the other hand git is good at keeping history.

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
If DEP 12 lands we are definitely going to need to start a "what do all the DEPs combined mean" document, but for now I think we can avoid it. I don't want to modify any DEPs in-place.

**Comment by carltongibson** [comment]
I've offered an alternate approach adding a "DEP 44" instead in #78. (I have no strong preference. :)

**Review by adamchainz — APPROVED** [review]
LGTM

I don’t mind how we merge this, even editing the past DEP. PEP’s change all the time, often with a little note like “a past version of this PEP recommended…”

**Comment by carltongibson** [comment]
OK, thanks for the feedback all. 

I think I'll review the DEP44 with minimal adjustments to DEP10 (just to point there) version and then ask for a vote on the forum to confirm. 

👍

**Comment by carltongibson** [comment]
Closing in favour of https://github.com/django/deps/pull/78.

**Pull request closed without merging**